grafana/resource_oncall_outgoing_webhook.go

```diff
@@ -0,0 +1,210 @@
+package grafana
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	onCallAPI "github.com/grafana/amixr-api-go-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceOnCallOutgoingWebhook() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+* [HTTP API](https://grafana.com/docs/grafana-cloud/oncall/oncall-api-reference/outgoing_webhooks/)
+`,
+		CreateContext: ResourceOnCallOutgoingWebhookCreate,
+		ReadContext:   ResourceOnCallOutgoingWebhookRead,
+		UpdateContext: ResourceOnCallOutgoingWebhookUpdate,
+		DeleteContext: ResourceOnCallOutgoingWebhookDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the outgoing webhook.",
+			},
+			"team_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the team.",
+			},
+			"webhook": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The url of the webhook.",
+			},
+			"data": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The data of the webhook.",
+			},
+			"user": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The auth data of the webhook.",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The auth data of the webhook.",
+			},
+			"authorization_header": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The authorization header.",
+			},
+			"forward_whole_payload": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Forwards whole payload of the alert to the webhook's url as POST data.",
+			},
+		},
+	}
+}
+
+func ResourceOnCallOutgoingWebhookCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*client).onCallAPI
+	if client == nil {
+		return diag.Errorf("grafana OnCall api client is not configured")
+	}
+
+	nameData := d.Get("name").(string)
+	teamIdData := d.Get("team_id").(string)
+	webhookData := d.Get("webhook").(string)
+	forwardWholePayloadData := d.Get("forward_whole_payload").(bool)
```

Comment:
```suggestion
	name := d.Get("name").(string)
	teamID := d.Get("team_id").(string)
	webhookURL := d.Get("webhook").(string)
	forwardWholePayload := d.Get("forward_whole_payload").(bool)
```

I think adding `data` here is a bit confusing since we also have a data field

Reply:
Same thing with all other attributes below, I'd remove the `data` suffix everywhere